pulp_smash/selectors.py

```diff
@@ -32,62 +33,122 @@
+        'Custom fields: {}'
+        .format(bug_json['issue']['id'], custom_field_id, custom_fields)
+    )
+
+
+def _convert_tpr(version_string):
+    """Convert a Target Platform Release (TPR) string to a ``Version`` object.
+
+    By default, a bug's TPR string is an empty string. It is unlikely to be set
+    until a fix has been implemented, and even then, it is quite possible that
+    the field will be left as an empty string. (Perhaps the user forgot to set
+    it, or set it to the wrong value.)
+
+    If ``version_string == ''``, this method pretends that ``version_string ==
+    '0'``. Why is this useful? Let's imagine that a bug has a status of
+    MODIFIED and a TPR of "": we can now assume that this bug is fixed in all
+    versions of Pulp. More generally, any time a bug is marked as fixed and no
+    TPR listed, we assume that the bug is fixed for all versions of Pulp.
 
+    :param version_string: A version string like "2.8.1" or "".
+    :returns: A ``packaging.version.Version`` object.
+    :raises: ``packaging.version.InvalidVersion`` if ``version_string`` is
+        invalid and not "".
+    """
+    if version_string == '':
+        return Version('0')
+    return Version(version_string)
+
+
+def _get_bug(bug_id):
+    """Fetch information about bug ``bug_id`` from https://pulp.plan.io.
+
+    Return a ``_Bug`` instance.
+    """
     # It's rarely a good idea to do type checking in a duck-typed language.
     # However, efficiency dictates we do so here. Without this type check, the
     # following will cause us to talk to the bug tracker twice and store two
     # values in the cache:
     #
-    #     _get_bug_status(1356)
-    #     _get_bug_status('1356')
+    #     _get_bug(1356)
+    #     _get_bug('1356')
     #
     if not isinstance(bug_id, int):
         raise TypeError(
             'Bug IDs should be integers. The given ID, {} is a {}.'
             .format(bug_id, type(bug_id))
         )
+
+    # Let's return the bug from the cache if possible. ¶ We shouldn't need to
+    # declare a global until we want to assign to it, but waiting causes Python
+    # itself to emit a SyntaxWarning.
+    global _BUG_STATUS_CACHE  # pylint:disable=global-variable-not-assigned
     try:
         return _BUG_STATUS_CACHE[bug_id]
     except KeyError:
         pass
 
-    # Get, cache and return bug status.
+    # The bug is not cached. Let's fetch, cache and return it.
     response = requests.get(
         'https://pulp.plan.io/issues/{}.json'.format(bug_id)
     )
     response.raise_for_status()
-    _BUG_STATUS_CACHE[bug_id] = response.json()['issue']['status']['name']
+    bug_json = response.json()
+    _BUG_STATUS_CACHE[bug_id] = _Bug(
+        bug_json['issue']['status']['name'],
+        _convert_tpr(_get_tpr(bug_json)),
+    )
     return _BUG_STATUS_CACHE[bug_id]
 
 
-def bug_is_testable(bug_id):
+def bug_is_testable(bug_id, pulp_version):
```

Comment:
Seems that I already got the answer for my question. The Version is being used everywhere and the Pulp Smash settings already returns a Version wrapped object. Changing this will require more refactoring and will not be so valuable to do.

That said let's follow the standard and always expect a `packaging.version.Version` object, this way one may not be confused if a string or a Version object should be the argument.